alibi_detect/saving/loading.py

```diff
@@ -527,7 +527,7 @@ def _replace(cfg: dict, orig: Optional[str], new: Optional[str]) -> dict:
     The updated dictionary.
     """
     for k, v in cfg.items():
-        if v == orig:
+        if isinstance(v == orig, bool) and v == orig:
```

Comment:
why this change?

Reply:
`_replace` loops through the entire `cfg`, which is a `dict` potentially containing numerous nested `dict`'s. For each value (`v`) in the `cfg` it compares it to `orig` and swaps it for `new` if it `v==orig`. 

The change to include the entire `self.meta` attribute in the config introduced an unexpected edge case for the above. `SpotTheDiffDrift` adds a number of non-standard items to `self.meta`:
https://github.com/SeldonIO/alibi-detect/blob/28aee232e8e121a30f1860814c6ea6f71419fb80/alibi_detect/cd/tensorflow/spot_the_diff.py#L161-L165

Some of these are `ndarray`'s, which broke the `if v == orig:` logic, since the `ndarray` `__eq__` returns an array. I've replaced it with the above so that we only check `v == orig` if `v == orig` returns a `bool`. This shouldn't affect the operation of `_replace` since it is a private function and we only use to for simple operations such as replacing `None` with `'None'`.

Reply:
It sounds like adding `self.meta` to config is potentially a bigger thing than the scope of the original issue? Perhaps we should reduce the scope here and re-think later (although if things appear ok then maybe it's good to press forward)?

What are the main benefits from adding `self.meta` to the config? Conscious that some of the information would be duplicated as `self.meta` may contain user-passed parameters which would be in the config anyway.

Reply:
The intention behind this was so that fields like `detector_type` could be viewed without needing to instantiate the detector. Based on offline discussion @jklaise and I have decided to remove `self.meta` from config for now, and will review at a later date (https://github.com/SeldonIO/alibi-detect/issues/567).